### PR TITLE
Search

### PR DESCRIPTION
**Pull request by piotrekinventica**
As a user I want to check if there is a measuring station in my nearby proximity

### CODE REVIEW

File: app/src/main/java/com/tenclouds/smogdog/ui/main/index/IndexViewModelImpl.kt [hunk 1]
```diff
@@ -39,12 +39,7 @@ constructor(private val model: IndexModel, navigator: Navigator, @PermissionObse
     }
 
     private val adapter = IndexAdapter()
-
-    var permissionDisposable: Disposable? = null
-        private set
-
-    var networkStateDisposable: Disposable? = null
-        private set
+    override val disposableList = mutableListOf<Disposable?>()
```
Comment (kubak89) [inline app/src/main/java/com/tenclouds/smogdog/ui/main/index/IndexViewModelImpl.kt]: Why the override?

File: app/src/main/java/com/tenclouds/smogdog/ui/base/AbstractViewModel.kt [hunk 2]
```diff
@@ -31,4 +35,19 @@ abstract class AbstractViewModel<V : MvvmView> : BaseObservable(), MvvmViewModel
     val isViewAttached: Boolean
         get() = view != null
 
+
+    fun checkPermission(permission: String): Boolean {
+        return navigator.checkPermission(permission)
+    }
+
+    fun requestPermission(vararg permissions: String) {
+        navigator.requestPermissions(arrayOf(*permissions), REQUEST_PERMISSION_LOCATION)
```
Comment (kubak89) [inline app/src/main/java/com/tenclouds/smogdog/ui/base/AbstractViewModel.kt]: Either REQUEST_PERMISSION_LOCATION should probably be passed to requestPermission method, so it's possible to differentiate between different permission request results.

File: app/src/main/java/com/tenclouds/smogdog/ui/main/index/IndexViewModel.kt [hunk 3]
```diff
@@ -6,18 +6,18 @@ import com.github.mikephil.charting.listener.OnChartValueSelectedListener
...
+    fun unsubscribeFromEverything()
     fun handlePermissionBroadcastIntent(intent: Intent)
-
     fun prepareQualityIndexData(data: QualityIndex, indexAdapter: IndexAdapter)
     fun getAdapter(): IndexAdapter
     fun getIndexValue(): Int
     fun getLineData(): ArrayList<Entry>
     fun getValueSelectedListener(): OnChartValueSelectedListener
     var screenState: IndexScreenState
+    val disposableList: MutableList<Disposable?>
```
Comment (kubak89) [inline app/src/main/java/com/tenclouds/smogdog/ui/main/index/IndexViewModel.kt]: Wouldn't CompositeDisposable be better here?

File: app/src/main/java/com/tenclouds/smogdog/ui/main/settings/SettingsViewModel.kt [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
 /**
  * Created by piotrkazmierczak on 01.08.2017.
  */
 
+@FragmentScope
 class SettingsViewModel
-@Inject constructor(val adapter: SettingsAdapter) : AbstractViewModel<SettingsView>() {
+@Inject constructor(val adapter: SettingsAdapter, navigator: Navigator) : AbstractViewModel<SettingsView>(navigator) {
 
 }
```
Comment (kubak89) [inline app/src/main/java/com/tenclouds/smogdog/ui/main/settings/SettingsViewModel.kt]: Empty class body.

File: app/src/main/java/com/tenclouds/smogdog/ui/main/MainViewModel.kt [hunk 5]
```diff
@@ -11,6 +12,6 @@ import javax.inject.Inject
 @ActivityScope
 class MainViewModel
 @Inject
-constructor(val adapter: MainAdapter) : AbstractViewModel<MainView>() {
+constructor(val adapter: MainAdapter, navigator: Navigator) : AbstractViewModel<MainView>(navigator) {
     fun getPagerAdapter() = adapter
```
Comment (kubak89) [inline app/src/main/java/com/tenclouds/smogdog/ui/main/MainViewModel.kt]: Unused method.

File: app/src/main/java/com/tenclouds/smogdog/ui/main/settings/SettingsViewModel.kt (hunk 4 above)
Comment (piotrekinventica) [inline app/src/main/java/com/tenclouds/smogdog/ui/main/settings/SettingsViewModel.kt]: Unrelated to this task. But yeah, according to YAGNI directives it shouldn't be in this merge request

File: app/src/main/java/com/tenclouds/smogdog/ui/main/MainViewModel.kt (hunk 5 above)
Comment (piotrekinventica) [inline app/src/main/java/com/tenclouds/smogdog/ui/main/MainViewModel.kt]: it is used, in databinding generated classes

**Pull request merged**